src/export/core.ts

```diff
@@ -93,11 +93,17 @@ type GetDependency<DependencyOption extends ComponentOption> = {
 
 type GetDependencies<MainType extends string> = GetDependency<Dependencies[Extract<MainType, DependenciesKeys>]>;
 
+// Retrieves the keys of a type that are not of type string or number.
+// https://github.com/microsoft/TypeScript/issues/25987#issuecomment-870515762
+type KnownKeys<T> = keyof {
+  [K in keyof T as string extends K ? never : number extends K ? never : K]: never
+};
+
 type ComposeUnitOption<OptionUnion extends ComponentOption> =
     // Will be never if some component forget to specify mainType.
     CheckMainType<GetMainType<OptionUnion>> &
     Omit<EChartsCoreOption, 'baseOption' | 'options'> & {
-        [key in GetMainType<OptionUnion>]?: Arrayable<
+        [key in KnownKeys<GetMainType<OptionUnion>>]?: Arrayable<
```

Comment:
Can you explain the detailed reason? I tested this change but it seems wrong.

| Before | After |
| :----: | :----: |
| ![image](https://github.com/apache/echarts/assets/26999792/d4a38a1c-d0f5-4795-a535-0f81dad47d3d) | ![image](https://github.com/apache/echarts/assets/26999792/8b6b8410-ca02-4ee5-8c34-8ad930757968) |
| | ![image](https://github.com/apache/echarts/assets/26999792/26788c67-d363-4062-a806-fdc28a0059db) |

Reply:
I wanted to provide an update regarding the PR that addressed the issue with the `vue-charts` library and the misalignment of EChartsOption with ComposeOption.

Upon further investigation, I realized that the problem does not originate from the ECharts library. Instead, it appears to be caused by a logic issue in another part of the codebase.

I sincerely apologize for any confusion or inconvenience this may have caused. After carefully considering it, I have decided to close this PR, since it is unnecessary to modify the ECharts types to resolve the issue.